### ISSUE TITLE
Support for TreeSummarizedExperiment

### ISSUE THREAD

**Issue by TuomasBorman** [nf-core/ampliseq#802]
### Description of feature

Hello!

Currently, the output options include `phyloseq`. In Bioconductor, there is emerging data container, `TreeSummarizedExperiment` (`TreeSE`), that harness the common `SummarizedExperiment` object. Support for `TreeSE` would benefit both `TreeSE` ecosystem and ampliseq community.

More information on `TreeSE` ecosystem you can find from here: https://microbiome.github.io/OMA/docs/devel/

-Tuomas

**Comment by d4straub** [comment]
Dear Tuomas,

that does indeed sound interesting, would you be able to open a PR with a module that produces such an object?

**Comment by TuomasBorman** [comment]
I can have a look, however, it might take some time due to other tasks. I will come back to you by the end of the year.

**Comment by hindrek** [comment]
Hi @TuomasBorman 

Thanks for the good initiative!

Have you considered creating a standalone module for TreeSummarizedExperiment, which could benefit not only nf-core/ampliseq but possibly other nf-core pipelines as well?

For example, SummarizedExperiment has a nf-core module, which can be easily integrated to any pipeline: https://nf-co.re/modules/summarizedexperiment_summarizedexperiment/

Hindrek

**Comment by TuomasBorman** [comment]
Hello @hindrek!

My knowledge on nf-core is unfortunately too little, but that sounds something that should be considered.

TreeSummarizedExperiment is versatile data container that can be used anywhere instead of phyloseq (metagenomic or amplicon data). TreeSE can be used also for other datatypes that have hierarchical structure that can be represented with tree. For data that do not include hierarchical structure, SummarizedExperiment would make sense (TreeSE inherits SE). It is the most widely used data container in Bioconductor, optimized for biological data in general that has abundance table.

@d4straub Should TreeSE be supported more widely in nf-core?

**Comment by d4straub** [comment]
Yes, a nf-core module would be optimal to reach potentially more users. However, it is pretty beneficial to first have working code before doing so, and that what we also create here by testing, imho. Putting TreeSE to nf-core also adds an overhead compared to a local module (when considering only nf--core/ampliseq development).

**Comment by d4straub** [comment]
Working code is implemented in the PR and was merged into dev. If anyone is interested in having that as nf-core module then I would recommend to open an issue there and link to #807.
